querier.hh

```diff
@@ -153,7 +134,7 @@ public:
         , _permit(std::move(permit))
         , _range(std::make_unique<const dht::partition_range>(std::move(range)))
         , _slice(std::make_unique<const query::partition_slice>(std::move(slice)))
-        , _reader(ms.make_reader(underlying_schema(), _permit, *_range, *_slice, pc, std::move(trace_ptr), streamed_mutation::forwarding::no, mutation_reader::forwarding::no))
+        , _reader(ms.make_reader(_schema, _permit, *_range, *_slice, pc, std::move(trace_ptr), streamed_mutation::forwarding::no, mutation_reader::forwarding::no))
```

Comment:
_schema is reversed for reversed reads, right?

If we're doing this, then we have to change all mutation_sources to accept a reversed schema for a reversed read, not just table::make_reader(), so that the querier can work with any mutation_source, not only the ones which are currently passed to it.